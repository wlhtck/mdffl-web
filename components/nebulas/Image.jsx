import styled from 'styled-components';
import { bool } from 'prop-types';

const StyledImage = styled.img`
  ${({ fullWidth }) => (fullWidth ? 'width: 100%;' : '')}
`;

StyledImage.propTypes = {
  fullWidth: bool,
};

StyledImage.defaultProps = {
  fullWidth: false,
};


export default StyledImage;
