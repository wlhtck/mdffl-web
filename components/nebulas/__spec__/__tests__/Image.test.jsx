import React from 'react';
import 'jest-styled-components';
import { render } from 'enzyme';
import Image from '../../Image';

describe('componenets/nebulas/Image', () => {
  describe('Image', () => {
    it('renders a <Image />', () => {
      expect(render(
        <Image
          src="https://images.unsplash.com/photo-1545906785-193d7758ba91?ixlib=rb-1.2.1&auto=format&fit=crop&w=600&q=80"
          alt="Bear"
        />,
      )).toMatchSnapshot();
    });

    it('renders a <Image fullWidth/>', () => {
      expect(render(
        <Image
          fullWidth
          src="https://images.unsplash.com/photo-1545906785-193d7758ba91?ixlib=rb-1.2.1&auto=format&fit=crop&w=600&q=80"
          alt="Bear"
        />,
      )).toMatchSnapshot();
    });
  });
});
