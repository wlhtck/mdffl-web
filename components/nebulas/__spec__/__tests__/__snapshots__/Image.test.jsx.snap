// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`componenets/nebulas/Image Image renders a <Image /> 1`] = `
<img
  alt="Bear"
  class=""
  src="https://images.unsplash.com/photo-1545906785-193d7758ba91?ixlib=rb-1.2.1&auto=format&fit=crop&w=600&q=80"
/>
`;

exports[`componenets/nebulas/Image Image renders a <Image fullWidth/> 1`] = `
.c0 {
  width: 100%;
}

<img
  alt="Bear"
  class="c0"
  src="https://images.unsplash.com/photo-1545906785-193d7758ba91?ixlib=rb-1.2.1&auto=format&fit=crop&w=600&q=80"
/>
`;
